File: Validation/valid input.cpp
#include <iostream>
#include <regex>
#include <string>
#include <vector>

using namespace std;

int main()
{

    string name;
    vector<string> validName;
    regex patternName("[A-z]+");

    while (true)
    {
        cout << "Enter the name : " << endl;
        cin >> name;

        if (regex_match(name, patternName))
        {
            validName.push_back(name);
            break;
        }

        else
        {
            cout << "Invalid name format. Please try again." << endl;
        }
    }

    string phone;
    vector<string> validPhone;
    regex patternPhone(R"([0-9]{3}\s?\-?[0-9]{4}\s?\-?[0-9]{4})");

    while (true)
    {
        cout << "Enter the Phone : " << endl;
        cin >> phone;

        if (regex_match(phone, patternPhone))
        {
            validPhone.push_back(phone);
            break;
        }

        else
        {
            cout << "Invalid Phone format. Please try again." << endl;
        }
    }

    string email;
    vector<string> validEmail;
    regex patternEmail(R"([A-z0-9\.]+\@[A-z0-9\.]+.[A-z0-9\.])");

    while (true)
    {
        cout << "Enter the Email : " << endl;
        cin >> email;

        if (regex_match(email, patternEmail))
        {
            validEmail.push_back(email);
            break;
        }

        else
        {
            cout << "Invalid email format. Please try again." << endl;
        }
    }

    cout << "Validation information : " << endl;

    cout << "Name : " << validName[0] << endl;

    cout << "Phone : " << validPhone[0] << endl;

    cout << "email : " << validEmail[0] << endl;

    return 0;
}

File: Validation/valid input.py
import re


def validation():

    name = input("Enter the name : ")
    nameValid = re.findall("[A-z]+", name)

    while not nameValid:
        print("Please enter a valid name")
        name = input("Enter the name : ")
        nameValid = re.findall("[A-z]+", name)

    phone = input("Enter the phone : ")
    phoneValid = re.findall(r"[0-9]{3}\s?\-?[0-9]{4}\s?\-?[0-9]{4}", phone)

    while not phoneValid:
        print("Please enter a valid phone")
        phone = input("Enter the phone : ")
        phoneValid = re.findall(r"[0-9]{3}\s?\-?[0-9]{4}\s?\-?[0-9]{4}", phone)

    email = input("Enter the email : ")
    emailValid = re.findall(r"[A-z0-9\.]+\@[A-z0-9\.]+.[A-z0-9\.]", email)

    while not emailValid:
        print("Please enter a valid email")
        phone = input("Enter the email : ")
        emailValid = re.findall(r"[A-z0-9\.]+\@[A-z0-9\.]+.[A-z0-9\.]+", email)

    name = "".join(nameValid)
    phone = "".join(phoneValid)
    email = "".join(emailValid)

    print("Validation information : ")
    print(f"Name : {name.capitalize()}")
    print(f"Phone : {phone}")
    print(f"email : {email}")


if __name__ == "__main__":
    validation()
